pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -220,6 +221,7 @@ func (c *clustersReconciler) Reconcile(ctx context.Context, req reconcile.Reques
 	}
 	if err != nil {
 		log.Error(err, "reading kubeconfig")
+		c.stopAndRemoveCluster(req.Name)
```

Comment:
This does not seem directly related, is it a drive-by fix to ensure there is no memory leak? Is there a unit test which would fail without it?